src/pages/index.js

```diff
@@ -1,130 +1,223 @@
-    galleryList
-  );
-  cardsSection.renderItems();
-
-  const editForm = document.forms["edit-form"];
-  const addForm = document.forms["add-form"];
-
-  const configuration = {
-    formSelector: ".modal__container",
-    inputSelector: ".modal__input",
-    submitButtonSelector: ".modal__save-button",
-    inactiveButtonClass: "transparent",
-    inputErrorClass: "popup__input_type_error",
-    errorClass: "popup__error_visible",
+  function toggleLikeButton(id, fillButton, vacateButton) {
+    api.toggleHeartIcon(id, fillButton, vacateButton).catch((error) => {
+      console.error("Caught an error: ", error.message);
+    });
+  }
+
+  // Elements
+  const imageClickHandler = ({ name, link }) => {
+    imagePopup.open({ name, link });
   };
 
-  const editValidator = new FormValidator(configuration, editForm);
-  const addValidator = new FormValidator(configuration, addForm);
+  const editValidator = new FormValidator(
+    Constants.configuration,
+    Constants.editForm
+  );
+  const addValidator = new FormValidator(
+    Constants.configuration,
+    Constants.addForm
+  );
+  const avatarValidator = new FormValidator(
+    Constants.configuration,
+    Constants.avatarForm
+  );
+  const user = new UserInfo(
+    Constants.profileName,
+    Constants.profileDescription,
+    Constants.avatarPhoto
+  );
 
   editValidator.enableValidation();
   addValidator.enableValidation();
-
-  const profile = document.querySelector("#profile");
-  const profileEditButton = profile.querySelector("#profile__edit_button");
-  const profileAddButton = profile.querySelector("#profile__add_button");
-  const profileName = profile.querySelector("#profile__name");
-  const profileDescription = profile.querySelector("#profile__description");
+  avatarValidator.enableValidation();
 
   const imagePopup = new PopupWithImage("#picture-modal");
 
   const editPopup = new PopupWithForm(
-    `#${editForm.parentElement.id}`,
+    `#${Constants.editForm.parentElement.id}`,
     handleProfileEditSubmit
   );
   const addPopup = new PopupWithForm(
-    `#${addForm.parentElement.id}`,
+    `#${Constants.addForm.parentElement.id}`,
     handleProfileAddSubmit
   );
-  const userInfo = new UserInfo(profileName, profileDescription);
 
   imagePopup.setEventListeners();
 
+  api
+    .getProfileInfo()
+    .then((results) => {
+      user.setUserInfo({ name: results.name, about: results.about });
+      user.setAvatar(results.avatar);
+    })
+    .catch((error) => {
+      console.error("Caught an error: ", error.message);
+    });
+
   //functions
+  const deletePopup = new ConfirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+    const items = [{ name: name, description: description }];
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        deletePopup.open();
+        deletePopup.setDefaultSaveButton();
```

Comment:
* You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.